salt/unit-test: Migrate to Python 3

gdemonet opened the pull request:
- removes `salttesting` dependency
- migrates away from deprecated `TestCase` methods
- migrates `Mock` objects call args checks

gdemonet committed:
salt/unit-test: Migrate to Python 3

- removes `salttesting` dependency
- migrates away from deprecated `TestCase` methods
- migrates `Mock` objects call args checks

TeddyAndrieux reviewed (approved):
Green build so it's ok for me

gdemonet committed:
salt: Prevent invalid escape sequence warnings

We need to pass `\;` to the `isoinfo` binary for reading file contents
from an ISO archive, but this escape sequence is not valid in Python,
and Python 3 warns about it.
We could have escaped the backslash, but instead rely on the raw string
notation for Python 3 (`r'my \escaped sequence'`).
